### PR TITLE
drivers: stm32 spi supports TI or Motorola standard protocols

### LINKED ISSUE #38683: stm32g0b1: spi: MOSI signal goes high before and after transmitting data
I noticed this issue while using the ws2812_spi LED strip driver on a nucleo_g0b1re board.

The led_ws2812 sample can be used to reproduce the issue. This [patch](https://github.com/simonguinot/zephyr/commit/3a9b09aced5643bd30f7ea2404401f2928c773ac) is needed to add support for the nucleo_g0b1re board to the led_ws2812 sample.

When running this sample, then the MOSI signal goes high for 80 microseconds before and after transmitting data:

stm32g0b1-spi1-mosi-high-before-after-transmission

stm32g0b1-spi1-mosi-high-zoom

The issue is that this high "preamble" state is interpreted as a high bit by the first WS2812 LED of the strip. And then the signal of all the LEDs is shifted by one bit, which results in visual glitches.

Note that from a SPI protocol standpoint, the MOSI signal is correct because the clock signal is off during this extra "high" states. Thus, this won't affect a real SPI device. But the WS2812 strip LED controller is not a SPI device. It doesn't use the clock signal. SPI is used here as a hack on a single wire (MOSI) to generate a control signal for the LEDs.

In addition, the issue is not systematic. It depends on the last data sent over MOSI. If the last 4 bytes are zero, then there are no high "preamble" state when sending the next buffer of data.

I also tried disabling DMA and interrupts but this doesn't affect the issue.

### PR DESCRIPTION
**Pull request by FRASTM**
This changes introduces a new property to the DTS of the stm32 spi to configure the Frame Format. 
Some STM32 devices, have a SPI interface which is compatible with the TI protocol. The FRF bit of the SPIx_CR2 register can be used to configure the SPI to be compliant with this protocol.
By default the frame format is motorola Mode.

To configure the TI mode, the dts should include the property as follows:
```
&spi1 {
	frame-format;
};

```

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/38683

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

File: dts/bindings/spi/st,stm32-spi-common.yaml [hunk 1]
```diff
@@ -12,6 +12,16 @@ properties:
...
+      type: boolean
+      required: false
+      description: |
+        Use Frame Format to support TI or motorola SPI mode when the SPI
+        interface is compatible with the TI protocol. The FRF bit of the
+        SPIx_CR2 configures the SPI to be compliant with this protocol.
+        false: SPI Motorola mode (default)
+        true: SPI TI mode
+
     pinctrl-0:
```
Comment (avisconti) [inline dts/bindings/spi/st,stm32-spi-common.yaml:25]: default is <false> for booleans, right?

**Review by avisconti — APPROVED** [review]
LGTM
(I assume the answer to my question is "yes")

**Review by simonguinot — APPROVED** [review]
I can confirm that adding the DT ```frame-format``` property to a SPI node fixes the [MOSI signal issue](https://github.com/zephyrproject-rtos/zephyr/issues/38683) for my test case.

In addition the patches are looking good to me.

Thanks @FRASTM.

**Comment by erwango** [comment]
@cfriedt, I think we've been too fast on this. It wasn't approved by any of the maintainers.

**Pull request merged**